src/main/java/org/sopt36/ninedotserver/mandalart/service/query/SubGoalQueryService.java

```diff
@@ -16,4 +24,18 @@ public class SubGoalQueryService {
     private final CoreGoalRepository coreGoalRepository;
     private final MandalartRepository mandalartRepository;
 
+    public List<SubGoalIdResponse> getSubGoalIds(Long userId, Long coreGoalId) {
+        CoreGoal coreGoal = coreGoalRepository.findById(coreGoalId)
+            .orElseThrow(() -> new SubGoalException(CORE_GOAL_NOT_FOUND));
+
+        if (!coreGoal.getMandalart().getUser().getId().equals(userId)) {
+            throw new SubGoalException(FORBIDDEN_ACCESS);
+        }
```

Comment:
요부분 함수로 빼면 더욱 좋을 것 같습니당! 그리고 getter를 연속으로 사용한다는 뜻은 이걸 해당 도메인 내부에 메서드로 구현할 수 있을 것 같아요!

```suggestion
public class Mandalart {
    private User user;
    // ... other fields

    public void ensureOwnedBy(Long userId) {
        if (!user.getId().equals(userId)) {
            throw new SubGoalException(FORBIDDEN_ACCESS);
        }
    }
}

public class CoreGoal {
    private Mandalart mandalart;
    // ...

    public void verifyUser(Long userId) {
        mandalart.ensureOwnedBy(userId);
    }
}

// 위의 두 부분을 추가하면 if 문 내의 로직은 다음과 같이 바뀔 수 있을 것 같습니다!
    coreGoal.verifyUser(userId);
```